driver-core/src/test/functional/com/mongodb/operation/AsyncQueryBatchCursorFunctionalSpecification.groovy

```diff
@@ -365,10 +365,9 @@ class AsyncQueryBatchCursorFunctionalSpecification extends OperationFunctionalSp
             }
 
             def futureResultCallback = new FutureResultCallback<BsonDocument>();
-            connection.commandAsync(getDatabaseName(), findCommand,
-                    NO_OP_FIELD_NAME_VALIDATOR, ReadPreference.secondaryPreferred(),
-                    CommandResultDocumentCodec.create(new DocumentCodec(), 'firstBatch'), binding.sessionContext,
-                    futureResultCallback)
+            connection.commandAsync(getDatabaseName(), findCommand, NO_OP_FIELD_NAME_VALIDATOR, ReadPreference.primary(),
```

Comment:
Use the connection source's binding rather than the one from the cached synchronous binding